sdk/core/core-https/src/policies/setClientRequestIdPolicy.ts

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { PipelineResponse, PipelineRequest, SendRequest } from "../interfaces";
+import { PipelinePolicy } from "../pipeline";
+
+/**
+ * The programmatic identifier of the setClientRequestIdPolicy.
+ */
+export const setClientRequestIdPolicyName = "setClientRequestIdPolicy";
+
+/**
+ * Each PipelineRequest gets a unique id upon creation.
+ * This policy passes that unique id along via an HTTP header to enable better
```

Comment:
This isn't directly related to your PR, but do you know if client request ids are meant to be unique to the SDK operation being called, or unique to the actual HTTP request being made? I'm curious what services expect when we perform retries.

Reply:
Hard to find clear guidance on this. Found this issue: https://github.com/Azure/azure-sdk/issues/576

I believe since this is a request id, each unique request we make should get its own id, hence it updates on retry, but maybe @bterlson knows better since he owns the above issue? ;)

Reply:
Talked to @johanste too... should be unique to the http request being made (and should not change between retries).

Reply:
Aren't those 2 statements at odds since a retry would create a new HTTP request? (I'm assuming you mean that it should be unique per operation invocation)

Reply:
So cloning a request should preserve it? I'll make the code change for that.

Oh and technically today we *do* preserve it in server requests, since retry happens after the header gets set by this policy, it's just the logs of the request object that show something different.